amazon-msk/cdk/README.WebStreaming.md

```diff
@@ -0,0 +1,411 @@
+
+
+### `roleName`: Zilla Plus EC2 Instance Assumed Role
+
+> Default: (generated)
+
+By default the deployment creates the Zilla Plus Role with the necessary roles and policies. If you prefer, you can specify your own role instead.
+
+List all IAM roles:
+
+```bash
+aws iam list-roles \
+  --query 'Roles[*].[RoleName,Arn]' \
+  --output table
+```
+
+Optionally override the assumed role (RoleName) for Zilla Plus via `cdk.context.json`, in the `WebStreaming` `roleName` variable.
+
+### `securityGroup`: Zilla Plus EC2 Instance Security Group
+
+> Default: (generated)
+
+By default the deployment creates the Zilla Plus Security Group with the necessary ports to be open. If you prefer, you can specify your own security group instead.
+
+List all security groups:
+
+```bash
+aws ec2 describe-security-groups \
+  --query 'SecurityGroups[*].[GroupId, GroupName]' \
+  --output table
+```
+
+Optionally override the security group IDs (GroupId) for Zilla Plus via `cdk.context.json`, in the `WebStreaming` `securityGroup` variable.
+
+### `cloudwatch` Zilla Plus variables
+
+> Default: (disabled)
+
+```json
+    "cloudwatch":
+    {
+        "logs":
+        {
+            "group": "<your cloudwatch log group name>"
+        },
+        "metrics":
+        {
+            "namespace": "<your cloudwatch metrics namespace>"
+        }
+    }
+```
+
+CloudWatch logging is enabled when you specify a log group name.
+CloudWatch metrics is enabled when you specify a metrics namespace.
+
+#### List All CloudWatch Log Groups
+
+```bash
+aws logs describe-log-groups \
+  --query 'logGroups[*].[logGroupName]' \
+  --output table
+```
+
+This command returns a table listing the names of all the log groups in your CloudWatch in the current AWS region.
+
+Optionally specify the CloudWatch Logs Group for Zilla Plus via `cdk.context.json`, in the `WebStreaming` `cloudwatch` `logs` `group` variable.
+
+#### List All CloudWatch Custom Metric Namespaces
+
+```bash
+aws cloudwatch list-metrics \
+  --query "Metrics[?\!contains(Namespace, 'AWS')].Namespace" \
+  --output table \
+| sort \
+| uniq
+```
+
+Optionally specify the CloudWatch Metrics Namespace for Zilla Plus via `cdk.context.json`, in the `WebStreaming` `cloudwatch` `metrics` `namespace` variable.
+
+### Enable JWT Access Tokens
+
+To enable the JWT authentication and API access control, you need to provide the `jwt` context variable. You will also need to set the JWT Issuer (`issuer`), JWT Audience (`audience`) and JWKS URL (`keys_url`) context variable inside the `jwt` object. Example:
+
+```json
+    "jwt": {
+      "issuer" : "https://auth.example.com",
+      "audience": "https://api.example.com",
+      "keysUrl": "https://{yourDomain}/.well-known/jwks.json"
+    }
+```
+
+### Enable Glue Schema Registry
+
+To enable the Glue Schema Registry for schema fetching, set the context variable `glueRegistry` to the name of the Glue Registry.
```

Comment:
To enable the Glue Schema Registry for schema fetching, set the context variable `glue` to the name of the Glue Registry.

```json
    "glue": {
      "registry" : "<your Glue Registry name>"
    }
```